File: hotel-domain/src/main/java/com/fran/hotel/domain/model/Room.java
package com.fran.hotel.domain.model;

import com.fasterxml.jackson.annotation.JsonTypeInfo;

@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
public record Room(
    String id,
    String hotelId,
    RoomType typeId,
    int floor,
    String number,
    String name,
    boolean isAvailable
) {
}

File: hotel-frontend/src/entities/room.ts
export type Room = {
  id: string;
  hotelId: string;
  number: string;
  typeId: string;
  floor: number;
  name: string;
  available: boolean;
};

export type NewRoomInput = {
  number: string;
  typeId: string;
  floor: number;
  name: string;
  available: boolean;
};

